core/input/input.cpp

```diff
@@ -1009,26 +1009,26 @@ void Input::joy_hat(int p_device, int p_val) {
 	map[HAT_LEFT].value = 0;
 
 	if (joy.mapping != -1) {
-		_get_mapped_hat_events(map_db[joy.mapping], 0, map);
+		_get_mapped_hat_events(map_db[joy.mapping], (HatDir)0, map);
```

Comment:
Should this be `HAT_UP` then?

Reply:
Functionally speaking, the behavior is identical. Semantically speaking, it depends if we want to say "We want the up direction" vs "We don't care which direction so let's just use the one with index zero". This seems to be a case of the latter, so I left this as `(HatDir)0`, but whether it should be `HAT_UP` instead is open for debate.